pkg/ebpf/btf.go

```diff
@@ -381,28 +382,25 @@ func (b *orderedBTFLoader) getEmbeddedBTF(platform btfPlatform, platformVersion,
 
 func (b *orderedBTFLoader) searchEmbeddedCollection(filename string) []string {
 	var matchingPaths []string
-	collectionTarball := filepath.Join(b.embeddedDir, "minimized-btfs.tar.xz")
-	err := archiver.NewTarXz().Walk(collectionTarball, func(f archiver.File) error {
-		th, ok := f.Header.(*tar.Header)
-		if !ok {
-			return fmt.Errorf("expected header to be *tar.Header but was %T", f.Header)
-		}
+	collectionTarball := filepath.Join(b.embeddedDir, btfArchiveName)
+	f, err := os.Open(collectionTarball)
+	if err != nil {
+		return nil
+	}
+	defer f.Close()
 
-		if !f.IsDir() {
-			if filepath.Base(th.Name) == filename {
-				pform := strings.Split(th.Name, string(os.PathSeparator))[0]
+	_ = archive.WalkTarXZArchive(collectionTarball, func(_ *tar.Reader, hdr *tar.Header) error {
```

Comment:
Looks like we are opening `collectionTarball` in this file at line 386, and then inside `WalkTarXZArchive` we are opening again. Is that expected?

Reply:
I see we are swallowing the error on purpose. Should we add a comment to explain why?

Reply:
> Is that expected?

That is a mistake. I refactored this code and this was probably left over. Fixing.